docs/tools.md

```diff
@@ -0,0 +1,114 @@
+# Tools
+
+Tools are Python packages that provide command-line interfaces. Tools can be invoked without installation using `uvx`, in which case their dependencies are installed in a temporary virtual environment isolated from the current project. Alternatively, tools can be installed with `uv tool install`, in which case their executables are placed in the `PATH` — an isolated virtual environment is still used but it is not treated as disposable.
+
+!!! note
+
+    See the [tools guide](./guides/tools.md) for an introduction to working with the tools interface — this document discusses details of tool management.
+
+## Tool environments
+
+Tools are installed into virtual environments which are created in the uv tools directory. When running tools with `uvx` or `uv tool run`, the virtual environments are stored in the uv cache directory and are treated as disposable.
+
+### Tools directory
+
+By default, the uv tools directory is named `tools` and is in the uv application state directory, e.g., `~/.local/share/uv/tools`. The location may be customized with the `UV_TOOL_DIR` environment variable.
+
+To display the path to the tool installation directory:
+
+```console
+$ uv tool dir
+```
+
+Tool environments are placed in a directory with the same name as the tool package, e.g., `.../tools/<name>`.
+
+### Mutating tool environments
+
+Tool environments are _not_ intended to be mutated directly. It is strongly recommended never to mutate a tool environment manually with a `pip` operation.
+
+Tool environments may be either mutated or re-created by subsequent `uv tool install` operations.
+
+To upgrade a single package in a tool environment:
+
+```
+$ uv tool install black --upgrade-package click
+```
+
+To upgrade all packages in a tool environment
+
+```
+$ uv tool install black --upgrade
+```
+
+To reinstall a single package in a tool environment:
+
+```
+$ uv tool install black --reinstall-package click
+```
+
+To reinstall all packages in a tool environment
+
+```
+$ uv tool install black --reinstall
+```
+
+All tool environment mutations will reinstall the tool executables, even if they have not changed.
+
+### Including additional dependencies
+
+Additional packages can be included during tool invocations and installations:
+
+```console
+$ uvx --with <extra-package> <tool-package>
+```
+
+```console
+$ uv tool install --with <extra-package> <tool-package>
+```
```

Comment:
Are these meant to be a demonstration of equivalent commands? Now I think I'm wondering how to choose whether to use `uvx` versus `uv tool install`.